### PR TITLE
Add dumptsmwal command to influx_inspect.

### PR DESCRIPTION
**Pull request by benbjohnson**
## Overview

This pull request adds the ability to dump the raw data from a TSM WAL file.

### Usage

```sh
$ influx_inspect dumptsmwal _00001.wal
```

###### Required only if applicable
- [x] Provide example syntax
- [x] Update man page when modifying a command
- [ ] [InfluxData Documentation](https://github.com/influxdata/docs.influxdata.com): issue filed or pull request submitted \<link to issue or pull request\>

### CODE REVIEW

**Comment by hercules-influx** [comment]
During a run of [megacheck](https://github.com/dominikh/go-tools) the following issues were discovered:
```/tmp/377385540/src/github.com/influxdata/influxdb/cmd/influx-tools/importer/importer.go:73:3: this value of rpi is never used (SA4006)
```

**Comment by hercules-influx** [comment]
During a run of [megacheck](https://github.com/dominikh/go-tools) the following issues were discovered:
```/tmp/256315518/src/github.com/influxdata/influxdb/cmd/influx-tools/importer/importer.go:73:3: this value of rpi is never used (SA4006)
```

**Comment by benbjohnson** [comment]
Added `-show-duplicates` flag to only show keys which have duplicate/out-of-order timestamps.

File: cmd/influx_inspect/dumptsmwal/dumptsmwal.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+		if err := cmd.process(path); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (cmd *Command) process(path string) error {
+	if filepath.Ext(path) != "."+tsm1.WALFileExtension {
+		log.Printf("invalid wal filename, skipping %s", path)
```
Comment (jacobmarble) [inline cmd/influx_inspect/dumptsmwal/dumptsmwal.go]: need a return here, else you aren't skipping anything

File: cmd/influx_inspect/dumptsmwal/dumptsmwal.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+			}
+			sort.Strings(keys)
+
+			for _, k := range keys {
+				for _, v := range entry.Values[k] {
+					t := v.UnixNano()
+
+					// Check for duplicate/out of order keys.
+					if min, ok := minTimestampByKey[k]; ok && t <= min {
+						minTimestampByKey[k] = t
```
Comment (jacobmarble) [inline cmd/influx_inspect/dumptsmwal/dumptsmwal.go]: `minTimestampByKey[k]` is not set unless the condition is true.
The condition is not true unless `minTimestampByKey[k]` has been set.

I haven't actually run the code, but I think this block is never hit.

**Pull request merged**